shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1899,20 +1886,39 @@ - (void)sendDeepLinkToFramework:(NSURL*)url completionHandler:(void (^)(BOOL suc
 // the wild, however, so I suspect that the API is built for a tvOS remote or
 // something, and perhaps only one ever appears in the set on iOS from a
 // keyboard.
+//
+// We define separate superPresses* overrides to avoid implicitly capturing self in the blocks
+// passed to the presses* methods below.
```

Comment:
There's an interesting (but maybe theoretical) question here: if handling a press both caused `self` to be released, but also claimed not to have handled the event (I'm not sure why that would happen, but I suspect you could write Dart code that would do it if you really wanted to), isn't the right thing to still forward the press to `super`?

If so, we actually would want the implicit capture. Since these are fire-once callbacks, capture isn't necessarily a problem. I did look at the usage though, and I can imagine a case where for some reason we never get the callback for some press due to an engine bug, and suddenly we have a leak that would be extremely hard to find.

This conversion makes sense in the context of the PR since the old code wasn't a strong ref, but since this structure and comment is making the non-retaining nature much more explicit I wanted to flag it as something the team may want to think about later.